source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py

```diff
@@ -155,118 +180,167 @@ def _extend_journey(pt_journey, fallback_dp, fallback_period_extremity):
     _extend_pt_sections_with_fallback_sections(pt_journey, aligned_fallback)
 
 
-def _build_from(
-    requested_orig_obj,
-    pt_journeys,
-    dep_mode,
-    streetnetwork_path_pool,
-    orig_accessible_by_crowfly,
-    orig_fallback_durations_pool,
-    request,
-):
+def _get_fallback_logic(fallback_type):
+    if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK:
+        return BeginningFallback()
 
-    accessibles_by_crowfly = orig_accessible_by_crowfly.wait_and_get()
-    fallback_durations = orig_fallback_durations_pool.wait_and_get(dep_mode)
+    if fallback_type == StreetNetworkPathType.ENDING_FALLBACK:
+        return EndingFallback()
 
-    for pt_journey in pt_journeys.journeys:
-        pt_origin = pt_journey.sections[0].origin
+    raise EntryPointException(
+        "Unknown fallback type : {}".format(fallback_type), response_pb2.Error.unknown_object
+    )
 
-        fallback_period_extremity = PeriodExtremity(pt_journey.departure_date_time, False)
-        # extend the journey with the fallback routing path
-        direct_path_type = StreetNetworkPathType.BEGINNING_FALLBACK
-        fallback_dp = streetnetwork_path_pool.wait_and_get(
-            requested_orig_obj, pt_origin, dep_mode, fallback_period_extremity, direct_path_type, request=request
-        )
 
-        if requested_orig_obj.uri != pt_origin.uri:
-            if pt_origin.uri in accessibles_by_crowfly.odt:
-                pt_journey.sections[0].origin.CopyFrom(requested_orig_obj)
-            elif _is_crowfly_needed(
-                pt_origin.uri, fallback_durations, accessibles_by_crowfly.crowfly, fallback_dp
-            ):
-                crowfly_departure_dt = (
-                    pt_journey.departure_date_time - fallback_durations[pt_origin.uri].duration
-                )
-                pt_journey.sections.extend(
-                    [
-                        _create_crowfly(
-                            pt_journey,
-                            requested_orig_obj,
-                            pt_origin,
-                            crowfly_departure_dt,
-                            pt_journey.sections[0].begin_date_time,
-                            dep_mode,
-                        )
-                    ]
-                )
-            else:
-                # extend the journey with the fallback routing path
-                _extend_journey(pt_journey, fallback_dp, fallback_period_extremity)
+class BeginningFallback:
+    def get_pt_section(self, pt_journey):
+        for s in pt_journey.sections:
+            if s.type == response_pb2.CROW_FLY:
```

Comment:
That's a fair point. But I actually want the first section which is not a crowfly (could be `public_transport` or `boarding`, `alighting` etc...). I'm renaming the function to avoid confusion !